### PR TITLE
fix(storage):  fix memory usage calculation for shared-buffer

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

We did not calculate the size of shared-buffer correctly. It will cause the wrong usage of memory show in grafana. And it may also make us can not limit the memory used by shared-buffer before compute-node OOM

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by TennyZhuang — APPROVED** [review]

**Comment by fuyufjh** [comment]
Consider merging this? https://github.com/risingwavelabs/risingwave/pull/9525, which also counts the overhead of jemalloc

File: src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs [hunk 1]
```diff
@@ -379,12 +379,15 @@ impl SharedBufferBatch {
         batch_items
             .iter()
             .map(|(k, v)| {
-                k.len() + {
-                    match v {
-                        HummockValue::Put(val) => val.len(),
-                        HummockValue::Delete => 0,
+                std::mem::size_of::<HummockValue<Bytes>>()
+                    + std::mem::size_of::<Bytes>()
```
Comment (yuhao-su) [inline src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs:383]: Looks like `size_of::<HummockValue<Bytes>>` already includes `size_of::<Bytes>`?

**Comment by fuyufjh** [comment]
Shall we close this? Or merge this or #9525?

**Comment by wenym1** [comment]
https://github.com/risingwavelabs/risingwave/pull/9525 seems better. +1 for https://github.com/risingwavelabs/risingwave/pull/9525

**Pull request closed without merging**